lexos/static/js/scripts_cut.js

```diff
@@ -1,382 +1,663 @@
+    // needsWarning = true; // For testing
+    if (needsWarning === true) {
+        $('#needsWarning').val('true')
+        const sizeWarning = 'Current cut settings will result in over 100 \
+        new segments. Please be patient if you continue.'
+        const footerButtons = '<button type="button" class="btn btn-default" \
+        id="warningContinue">Continue Anyway</button> <button type="button" \
+        class="btn btn-default" data-dismiss="modal">Cancel</button>'
+        $('#warning-modal-footer').html(footerButtons)
+        $('#warning-modal-message').html(sizeWarning)
+        // Hide the processing icon and show the modal
+        $('#status-prepare').css({'visibility': 'hidden'})
+        $('#warning-modal').modal()
     }
-    // Get a list of each of the cutset indices
-    var listindex = indivdivs.index(this)
-    currID = activeFileIDs[listindex] // activeFileIDs is defined in the template file
-    var isCutByMS = $('.indivMS', this).is(':checked') // True if cut by milestone checked
-    // If not cut by milestone and no segment size, add to no individual cutsets array
-    if (!isCutByMS && thisCutVal == '') {
-      eltswithoutindividualopts.push(listindex)
+    else {
+        $('#needsWarning').val('false')
     }
-    // If no segment size
-    if (thisCutVal != '') {
-      // Get segment cut type
-      var thisCutType = $("input[name='cutType_" + currID + "']:checked").val()
-      // If not cut by milestone, use num_ variables set in template file
-      if (!(isCutByMS)) {
-        // Needs warning...
-        // If the number of characters-overlap size/segment size-overlap size > 100
-        if (thisCutType == 'letters' && (numChar[listindex] - thisOverVal) / (thisCutVal - thisOverVal) > maxSegs) {
-          needsWarning = true
-          // Same for segments and lines
-        } else if (thisCutType == 'words' && (numWord[listindex] - thisOverVal) / (thisCutVal - thisOverVal) > maxSegs) {
-          needsWarning = true
-        } else if (thisCutType == 'lines' && (numLine[listindex] - thisOverVal) / (thisCutVal - thisOverVal) > maxSegs) {
-          needsWarning = true
-          // Or if the segment size > 100
-        } else if (thisCutVal > maxSegs && eltswithoutindividualopts.length > 0) {
-          needsWarning = true
+} // end checkForWarnings
+
+
+// checkMilestone
+/**
+ * checks milestone for warnings
+ * @param {number} maxSegs - maximum number of segments
+ * @param {string} defCutTypeValue - cut type
+ * @param {number} cutVal - segment size
+ * @param {number} overVal - overlap size
+ * @param {array} eltswithoutindividualopts - elements without cutsets
+ * @returns {boolean}
+ */
+function checkMilestone (maxSegs, defCutTypeValue, cutVal,
+                         overVal, eltswithoutindividualopts) {
+    // If cut by milestone is checked
+    if ($('input[name=\'cutByMS\']:checked').length === 0) {
+        // For cutting by characters
+        if (defCutTypeValue === 'letters') {
```

Comment:
Is it possible to flatten this nested `if` statement @Weiqi97 maybe you can help them a little with this.